tests/integration/custom_metadata_test.py

```diff
@@ -1,132 +1,855 @@
+    Table,
+    AtlasGlossaryTerm,
+    AtlasGlossary,
+)
+from pyatlan.model.core import CustomMetadata, to_snake_case
+from pyatlan.model.enums import AtlanCustomAttributePrimitiveType, AtlanTypeCategory
+from pyatlan.model.search import Term, DSL, Bool, IndexSearchRequest, Exists
+from pyatlan.model.typedef import AttributeDef, CustomMetadataDef, EnumDef
+from tests.integration.glossary_test import GlossaryTest
+
+PREFIX = "psdk2-"
+CM_PREFIX = PREFIX + "CM"
+FIXED_USER = "ernest"
+
+CM_RACI = PREFIX + "RACI"
+CM_IPR = PREFIX + "IPR"
+CM_QUALITY = PREFIX + "DQ"
+
+CM_ATTR_RACI_RESPONSIBLE = "Responsible"
+CM_ATTR_RACI_ACCOUNTABLE = "Accountable"
+CM_ATTR_RACI_CONSULTED = "Consulted"
+CM_ATTR_RACI_INFORMED = "Informed"
+CM_ATTR_RACI_EXTRA = "Extra"
+
+CM_ATTR_RACI_RESPONSIBLE_RENAMED = to_snake_case(CM_ATTR_RACI_RESPONSIBLE)
+CM_ATTR_RACI_ACCOUNTABLE_RENAMED = to_snake_case(CM_ATTR_RACI_ACCOUNTABLE)
+CM_ATTR_RACI_CONSULTED_RENAMED = to_snake_case(CM_ATTR_RACI_CONSULTED)
+CM_ATTR_RACI_INFORMED_RENAMED = to_snake_case(CM_ATTR_RACI_INFORMED)
+CM_ATTR_RACI_EXTRA_RENAMED = to_snake_case(CM_ATTR_RACI_EXTRA)
+
+CM_ATTR_IPR_LICENSE = "License"
+CM_ATTR_IPR_VERSION = "Version"
+CM_ATTR_IPR_MANDATORY = "Mandatory"
+CM_ATTR_IPR_DATE = "Date"
+CM_ATTR_IPR_URL = "URL"
+
+CM_ATTR_IPR_LICENSE_RENAMED = to_snake_case(CM_ATTR_IPR_LICENSE)
+CM_ATTR_IPR_VERSION_RENAMED = to_snake_case(CM_ATTR_IPR_VERSION)
+CM_ATTR_IPR_MANDATORY_RENAMED = to_snake_case(CM_ATTR_IPR_MANDATORY)
+CM_ATTR_IPR_DATE_RENAMED = to_snake_case(CM_ATTR_IPR_DATE)
+CM_ATTR_IPR_URL_RENAMED = to_snake_case(CM_ATTR_IPR_URL)
 
-CM_NAME = "psdk-CustomMetadataTest"
-CM_ATTR_LICENSE = "License"
-CM_ATTR_VERSION = "Version"
-CM_ATTR_MANDATORY = "Mandatory"
-CM_ATTR_DATE = "Date"
-CM_ATTR_URL = "URL"
+CM_ATTR_QUALITY_COUNT = "Count"
+CM_ATTR_QUALITY_SQL = "SQL"
+CM_ATTR_QUALITY_TYPE = "Type"
+CM_ENUM_DQ_TYPE = PREFIX.replace("-", "_") + "DataQualityType"
+DQ_TYPE_LIST = [
+    "Accuracy",
+    "Completeness",
+    "Consistency",
+    "Timeliness",
+    "Validity",
+    "Uniqueness",
+]
 
-# NOTE: Tests are intended to run in the order specified in this file,
-# to complete creation prior to testing prior to deletion.
+CM_ATTR_QUALITY_COUNT_RENAMED = to_snake_case(CM_ATTR_QUALITY_COUNT)
+CM_ATTR_QUALITY_SQL_RENAMED = to_snake_case(CM_ATTR_QUALITY_SQL)
+CM_ATTR_QUALITY_TYPE_RENAMED = to_snake_case(CM_ATTR_QUALITY_TYPE)
 
+GROUP_NAME1 = PREFIX + "1"
+GROUP_NAME2 = PREFIX + "2"
 
-@pytest.fixture
+_removal_epoch: Optional[int]
+
+
+@pytest.fixture(scope="session")
 def client() -> AtlanClient:
     return AtlanClient()
 
 
-def test_001_create_custom_metadata(client: AtlanClient):
-    attribute_defs = [
-        AttributeDef(
-            name=CM_ATTR_LICENSE,
-            display_name=CM_ATTR_LICENSE,
-            type_name="string",
-            options=AttributeDef.Options(),
+@pytest.fixture(scope="module")
+def cm_ipr(client: AtlanClient) -> Generator[CustomMetadataDef, None, None]:
```

Comment:
The presence of all the asserts makes this look like a test not a fixture. Because it's a fixture if any of the asserts before the yield fail then I don't believe that the code after the yield will be run.